pkg/client/logs.go

```diff
@@ -151,47 +208,62 @@ func (s *SonobuoyClient) LogReader(cfg *LogConfig) (*Reader, error) {
 	if err != nil {
 		return nil, err
 	}
+	podCh := make(chan *v1.Pod)
+	var wg sync.WaitGroup
 
-	pods, err := getPodsForLogs(client, cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	// We must make sure the error channel has capacity enough so that it never blocks
-	// in case of early exits.
-	numContainers := 0
-	for _, pod := range pods.Items {
-		numContainers += len(pod.Spec.Containers)
+	if cfg.Follow {
+		// Extra waitGroup item ensures we just keep processing forever.
+		wg.Add(1)
+		if err := watchPodsToStreamLogs(client, cfg, podCh); err != nil {
+			return nil, errors.Wrap(err, "failed to watch for new pods")
+		}
+	} else {
+		if err := getPodsToStreamLogs(client, cfg, podCh); err != nil {
+			return nil, errors.Wrap(err, "failed to list pods")
+		}
 	}
 
-	errc := make(chan error, numContainers)
+	// TODO figure out this error channel logic; we dont want to get a deadlock and we dont know
```

Comment:
Are you happy that we avoid deadlock here and could remove this TODO comment?

Reply:
Yes only applied when we weren't reading from the `errc` at the same time; the old logic just drained the bytestream before ever reading from the errc once.